vm/src/pyobject.rs

```diff
@@ -697,6 +801,7 @@ impl PyContext {
 pub struct PyObject {
     pub payload: PyObjectPayload,
     pub typ: Option<PyObjectRef>,
+    weak_refs: Vec<PyObjectWeakRef>,
```

Comment:
that's quite a heavy approach. In CPython a weakref support is optional (signaled at type object level), and disabled for most builtin types to save on memory and time.
Guessing that's related to #371 too though.